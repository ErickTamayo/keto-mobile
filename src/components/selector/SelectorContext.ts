import { createContext } from 'react'

export interface SelectorContext {
  selectedValue?: string
  onSelect?: (value: string) => void
}

export const { Provider, Consumer } = createContext({})
