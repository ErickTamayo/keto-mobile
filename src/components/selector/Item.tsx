import React, { Component } from 'react'
import { Text, View, TouchableOpacity } from 'react-native'
import { Consumer, SelectorContext } from './SelectorContext'
import st from '../../styles'

export interface Props {
  label: string
  value: string
}

export default class Item extends Component<Props, {}> {
  render() {
    const { label, value } = this.props

    return (
      <Consumer>
        {({ selectedValue, onSelect }: SelectorContext) => (
          <TouchableOpacity
            onPress={() => onSelect && onSelect(value)}
            disabled={value === selectedValue}
          >
            <View
              style={[
                st.border.b3,
                st.justify.center,
                st.m.l4,
                st.m.r4,
                st.minWidth.w5,
                value === selectedValue ? st.border.purple : st.border.transparent,
              ]}
            >
              <Text
                style={[
                  st.font.medium,
                  st.text.lg,
                  st.text.center,
                  value === selectedValue ? st.text.purple : st.text.greyLight,
                ]}
              >
                {label}
              </Text>
            </View>
          </TouchableOpacity>
        )}
      </Consumer>
    )
  }
}
