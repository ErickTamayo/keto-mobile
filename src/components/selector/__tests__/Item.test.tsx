import 'react-native'
import React from 'react'
import { TouchableOpacity, GestureResponderEvent } from 'react-native'
import { mount } from 'enzyme'
import Item, { Props } from '../Item'
import { Provider } from '../SelectorContext'

describe('<Item />', () => {
  it('renders correctly', () => {
    const props: Props = {
      label: 'item-label',
      value: 'item-value',
    }
    const wrapper = mount(<Item {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should be highlighted once selectedValue is the same as the item value', () => {
    const props: Props = {
      label: 'item-label',
      value: 'item-value',
    }

    const onSelect = jest.fn()

    const wrapper = mount(
      <Provider value={{ selectedValue: 'item-value', onSelect }}>
        <Item {...props} />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('calls onSelect once pressed', () => {
    const props: Props = {
      label: 'item-label',
      value: 'item-value',
    }

    const onSelect = jest.fn()

    const wrapper = mount(
      <Provider value={{ selectedValue: 'item-value', onSelect }}>
        <Item {...props} />
      </Provider>
    )

    wrapper.find(TouchableOpacity).prop('onPress')!({} as GestureResponderEvent)
    expect(onSelect).toBeCalled()
    expect(onSelect.mock.calls).toEqual([['item-value']])
  })
})
