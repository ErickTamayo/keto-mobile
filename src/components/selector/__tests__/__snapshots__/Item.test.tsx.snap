// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders correctly 1`] = `
<Item
  label="item-label"
  value="item-value"
>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    onPress={[Function]}
  >
    <AnimatedComponent
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Component
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Component
            style={
              Array [
                Object {
                  "borderBottomWidth": 3,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 26,
                },
                Object {
                  "marginRight": 26,
                },
                Object {
                  "minWidth": 40,
                },
                Object {
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 3,
                  },
                  Object {
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 26,
                  },
                  Object {
                    "marginRight": 26,
                  },
                  Object {
                    "minWidth": 40,
                  },
                  Object {
                    "borderColor": "transparent",
                  },
                ]
              }
            >
              <Component
                style={
                  Array [
                    Object {
                      "fontFamily": "avenir-next-medium",
                    },
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#cfcfcf",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "avenir-next-medium",
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#cfcfcf",
                      },
                    ]
                  }
                >
                  item-label
                </Text>
              </Component>
            </View>
          </Component>
        </View>
      </Component>
    </AnimatedComponent>
  </TouchableOpacity>
</Item>
`;

exports[`<Item /> should be highlighted once selectedValue is the same as the item value 1`] = `
<Item
  label="item-label"
  value="item-value"
>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={true}
    onPress={[Function]}
  >
    <AnimatedComponent
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Component
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Component
            style={
              Array [
                Object {
                  "borderBottomWidth": 3,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 26,
                },
                Object {
                  "marginRight": 26,
                },
                Object {
                  "minWidth": 40,
                },
                Object {
                  "borderColor": "#724bb7",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 3,
                  },
                  Object {
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 26,
                  },
                  Object {
                    "marginRight": 26,
                  },
                  Object {
                    "minWidth": 40,
                  },
                  Object {
                    "borderColor": "#724bb7",
                  },
                ]
              }
            >
              <Component
                style={
                  Array [
                    Object {
                      "fontFamily": "avenir-next-medium",
                    },
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#724bb7",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "avenir-next-medium",
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#724bb7",
                      },
                    ]
                  }
                >
                  item-label
                </Text>
              </Component>
            </View>
          </Component>
        </View>
      </Component>
    </AnimatedComponent>
  </TouchableOpacity>
</Item>
`;
